Source/VMware.vSphereDSC/DSCResources/VMHostAccount.ps1

```diff
@@ -0,0 +1,279 @@
+        $this.PopulateVMHostAccountParams($vmHostAccountParams, $this.DescriptionParameterName, $this.Description)
+
+        return $vmHostAccountParams
+    }
+
+    <#
+    .DESCRIPTION
+
+    Creates a new Permission with the passed Role for the specified VMHost Account.
+    #>
+    [void] CreateAccountPermission($vmHostAccount) {
+        $accountRole = Get-VIRole -Server $this.Connection -Name $this.Role -ErrorAction SilentlyContinue
+        if ($null -eq $accountRole) {
+            throw "The passed role $($this.Role) is not present on the server."
+        }
+
+        try {
+            New-VIPermission -Server $this.Connection -Entity $this.Server -Principal $vmHostAccount -Role $accountRole -ErrorAction Stop
+        }
+        catch {
+            throw "Cannot assign role $($this.Role) to account $($vmHostAccount.Name). For more information: $($_.Exception.Message)"
+        }
+    }
+
+    <#
+    .DESCRIPTION
+
+    Creates a new VMHost Account with the specified properties.
+    #>
+    [void] AddVMHostAccount() {
+        $vmHostAccountParams = $this.GetVMHostAccountParams()
+        $vmHostAccountParams.Id = $this.Id
+
+        $vmHostAccount = $null
+
+        try {
+            $vmHostAccount = New-VMHostAccount @vmHostAccountParams
+        }
+        catch {
+            throw "Cannot create VMHost Account $($this.Id). For more information: $($_.Exception.Message)"
+        }
+
+        $this.CreateAccountPermission($vmHostAccount)
+    }
+
+    <#
+    .DESCRIPTION
+
+    Updates the VMHost Account with the specified properties.
+    #>
+    [void] UpdateVMHostAccount($vmHostAccount) {
+        $vmHostAccountParams = $this.GetVMHostAccountParams()
+
+        try {
+            $vmHostAccount | Set-VMHostAccount @vmHostAccountParams
+        }
+        catch {
+            throw "Cannot update VMHost Account $($this.Id). For more information: $($_.Exception.Message)"
+        }
+
+        if ($this.ShouldCreateAcountPermission($vmHostAccount)) {
+            $this.CreateAccountPermission($vmHostAccount)
+        }
+    }
+
+    <#
+    .DESCRIPTION
+
+    Removes the VMHost Account.
+    #>
+    [void] RemoveVMHostAccount($vmHostAccount) {
+        try {
+            $vmHostAccount | Remove-VMHostAccount -Server $this.Connection -Confirm:$false -ErrorAction Stop
+        }
+        catch {
+            throw "Cannot remove VMHost Account $($this.Id). For more information: $($_.Exception.Message)"
+        }
+    }
+
+    <#
+    .DESCRIPTION
+
+    Populates the result returned from the Get() method with the values of the VMHost Account from the server.
+    #>
+    [void] PopulateResult($vmHostAccount, $result) {
+        if ($null -ne $vmHostAccount) {
+            $permission = Get-VIPermission -Server $this.Connection -Entity $this.Server -Principal $vmHostAccount -ErrorAction SilentlyContinue | Where-Object { $_.Role -eq $this.Role }
```

Comment:
Filterinf with Role here is tricky. Does that mean there could be multiple permission for the account and you are filtering to find the right one? From the reource interface I'm with the impression that only one permission should present for one account. If releation is one to many then Role could be of array type.

Reply:
Yes filtering here is not necessary because one account can have only one permission on the same entity. I will remove the filtering part from the implementation as it is redundant.